source/jormungandr/tests/line_sections_tests.py

```diff
@@ -165,90 +248,181 @@ def test_traffic_reports_on_stop_areas(self):
-        )
+        scenario = {
+            'line:1': True,
+            # 'line:2': False,
+            'line:3': False,
+        }
 
-        # There is a disruption and it affects 1 stop of the line
-        # whereas the line itself is not concerned (the
-        # disrupted routes are linked to other lines)
-        #
-        # This test doesn't work with the actual behavior
-        # assert not self.has_tf_disruption(
-        #     'lines/line:2/traffic_reports',
-        #     'line_section_on_line_1_other_effect',
-        # )
-
-        # At least, the disruption and the line are not linked
+        for line, result in scenario.iteritems():
+            assert result == self.has_tf_disruption(
+                'lines/{}/traffic_reports'.format(line), 'line_section_on_line_1_other_effect'
+            )
+            assert result == self.has_tf_linked_disruption(
+                'lines/{}/traffic_reports'.format(line),
+                'line_section_on_line_1_other_effect',
+                ObjGetter('lines', line),
+            )
+
+        # there is no link between line:2 and the disruption
         assert not self.has_tf_linked_disruption(
             'lines/line:2/traffic_reports', 'line_section_on_line_1_other_effect', ObjGetter('lines', 'line:2')
         )
 
+        # line_section_on_line_2
+        scenario = {
+            # 'line:1': False,
+            'line:2': True,
+            'line:3': False,
+        }
+
+        for line, result in scenario.iteritems():
+            assert result == self.has_tf_disruption(
+                'lines/{}/traffic_reports'.format(line), 'line_section_on_line_2'
+            )
+            assert result == self.has_tf_linked_disruption(
+                'lines/{}/traffic_reports'.format(line), 'line_section_on_line_2', ObjGetter('lines', line)
+            )
+
+        # there is no link between line:1 and the disruption
+        assert not self.has_tf_linked_disruption(
+            'lines/line:1/traffic_reports', 'line_section_on_line_2', ObjGetter('lines', 'line:1')
+        )
+
     def test_traffic_reports_on_routes(self):
         """
         for routes since we display the impacts on all the stops (but we do not display a route object)
         we display the disruption even if the route has not been directly impacted
         """
 
-        def has_dis(q):
-            r = self.default_query(q)
-            return 'line_section_on_line_1' in (d['disruption_id'] for d in r['disruptions'])
+        # line_section_on_line_1
+        # route 3 has been impacted by the line section but it has no stoppoint
+        # in the line section so in this case we do not display the disruption
+        scenario = {
+            'route:line:1:1': True,
+            'route:line:1:2': False,
+            'route:line:1:3': False,
+            'route:line:2:1': False,
+            'route:line:3:1': True,
```

Comment:
Shouldn't it be False here ? (commented if we follow the what's already done elsewhere)

Reply:
I'm not really sure if it should be False because of the note below the function : 

for routes since we display the impacts on all the stops (but we do not display a route object) we display the disruption even if the route has not been directly impacted

But I don't mind commenting the lines if you think it is not the right behavior anyway.

Reply:
As discussed:
* the current behavior si OK for now (at least for Kisio) but would be improved if routes where managed.
* let's try to write tests for the targeted behavior but add a clear xfail for them
  * If it's heavy, let's switch back to targeted behavior but commented tests
  * If possible let's make those xfail strict so that we maintain them (like we should do for artemis I guess :sweat_smile:  )